src/features/positions/components/LpPositionListItem.tsx

```diff
@@ -12,91 +11,51 @@ import { ButtonPressAnimation } from '@/components/animations';
 import { useBackendNetworksStore } from '@/state/backendNetworks/backendNetworks';
 
-function getRangeStatus(assets: RainbowUnderlyingAsset[], isConcentratedLiquidity: boolean) {
-  if (!isConcentratedLiquidity) {
-    return 'full_range';
-  }
-
-  const isOutOfRange =
-    assets.some(asset => {
-      return asset.quantity === '0';
-    }) || assets.length === 1;
-
-  return isOutOfRange ? 'out_of_range' : 'in_range';
-}
-
 type Props = {
   assets: RainbowUnderlyingAsset[];
   value: { amount: string; display: string };
-  isConcentratedLiquidity: boolean;
+  rangeStatus: RangeStatus;
+  allocation: LpAllocation;
   dappVersion?: string;
   name?: string;
   onPress?: (asset: RainbowUnderlyingAsset['asset']) => void;
 };
 
-export const LpPositionListItem: React.FC<Props> = React.memo(({ assets, value, isConcentratedLiquidity, dappVersion, name, onPress }) => {
+export const LpPositionListItem: React.FC<Props> = React.memo(({ assets, value, rangeStatus, allocation, dappVersion, name, onPress }) => {
   const { colors } = useTheme();
   const separatorSecondary = useForegroundColor('separatorSecondary');
   const chainsNativeAsset = useBackendNetworksStore(state => state.getChainsNativeAsset());
 
-  const rangeStatus = useMemo(() => getRangeStatus(assets, isConcentratedLiquidity), [assets, isConcentratedLiquidity]);
-
-  const { displayAssets, allocationPercentages, orderedAssets, hasOthers } = useMemo(() => {
-    // For pools with >2 assets, show top 2 + "Other"
-    // Assets are already sorted by value in the parser
+  const { displayAssets, orderedAssets } = useMemo(() => {
+    // For pools with >2 assets, show first 2 + "Other"
     const orderedAssets = assets.length > 2 ? assets.slice(0, 2) : assets;
-    const hasOthers = assets.length > 2;
     let displayAssets = orderedAssets;
 
-    // Calculate allocation percentages
-    let allocationPercentages = displayAssets.map(asset => {
-      if (value.amount === '0') {
-        return asset.quantity === '0' ? 0 : 100;
-      }
-      return Math.round(parseFloat(divide(asset.value.amount, value.amount)) * 100);
-    });
-
-    // If native/wrapped asset comes first and split is 50/50, flip the order so non-native token is displayed first
-    if (displayAssets.length === 2 && allocationPercentages[0] === 50 && allocationPercentages[1] === 50) {
+    // If native/wrapped asset comes first, flip the order so non-native token is displayed first
+    if (allocation.splits === 2) {
       const firstSymbol = displayAssets[0].asset.symbol?.toLowerCase();
       const nativeAsset = chainsNativeAsset[displayAssets[0].asset.chainId];
       const nativeSymbol = nativeAsset?.symbol?.toLowerCase();
 
       if (firstSymbol && nativeSymbol && (firstSymbol === nativeSymbol || firstSymbol === `w${nativeSymbol}`)) {
         displayAssets = [displayAssets[1], displayAssets[0]];
-        allocationPercentages = [allocationPercentages[1], allocationPercentages[0]];
```

Comment:
Avoiding reversing `0% / 100%` (like we used to) to line-up with the backend-driven pool symbol ordering